### PR TITLE
Added models for hotel loader

### PR DESCRIPTION
**Pull request by PKorovinPromo-Z**

### CODE REVIEW

File: HappyTravel.FastpayhotelsConnector.Common/Constants.cs [hunk 1]
```diff
@@ -11,4 +11,5 @@ public static class Constants
     public const string ApiVersion = "1";
     public const string FastpayhotelsTokenClient = "FastpayhotelsTokenClient";
     public static readonly TimeSpan AuthTokenLifeTimeDays = TimeSpan.FromDays(400); // The access token has a validity of 400 days 
+    public static readonly string[] Languages = new[] { "ar", "en", "ru", "bg", "de", "el", "es", "fr", "it", "hu", "pl", "pt", "ro", "sr", "tr" };
```
Comment (suhan3z) [inline HappyTravel.FastpayhotelsConnector.Common/Constants.cs:14]: Can't we use ISO codes here? like CultureInfo.TwoLetterISOLanguageName

File: HappyTravel.FastpayhotelsConnector.Common/Models/Catalogue/Caption.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+public class Caption
+{
+    /// <summary>
+    /// Caption based on culture code
+    /// </summary>
+    public string Content { get; set; }
+    /// <summary>
+    /// Culture code
+    /// </summary>
+    public string Lang { get; set; }
```
Comment (suhan3z) [inline HappyTravel.FastpayhotelsConnector.Common/Models/Catalogue/Caption.cs:12]: Better to use Culture

File: HappyTravel.FastpayhotelsConnector.Common/Models/Catalogue/CatalogueRequest.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿namespace HappyTravel.FastpayhotelsConnector.Common.Models.Catalogue;
+
+public class CatalogueRequest
+{
+    /// <summary>
+    /// Filter data by modificacition date, this filter is optional. Format ISO 8601
+    /// </summary>
+    public string LastUpdateDate { get; set; }
+    /// <summary>
+    /// List of languages in which the information is requested, use culture ids based on ISO 639-1, this filter is optional. If languages is null or empty by default system return "English".
```
Comment (suhan3z) [inline HappyTravel.FastpayhotelsConnector.Common/Models/Catalogue/CatalogueRequest.cs:10]: Same here

File: HappyTravel.FastpayhotelsConnector.Common/Models/DataObject.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿namespace HappyTravel.FastpayhotelsConnector.Common.Models;
+
+public class DataObject
```
Comment (suhan3z) [inline HappyTravel.FastpayhotelsConnector.Common/Models/DataObject.cs:3]: Too wide name, maybe FastPayObject or something like this?

File: HappyTravel.FastpayhotelsConnector.Common/Models/Hotel/Description.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+public class Description
+{
+    /// <summary>
+    /// Define type of description, long or short
+    /// </summary>
+    public string Type { get; set; }
+    /// <summary>
+    /// Culture code
+    /// </summary>
+    public string Language { get; set; }
```
Comment (suhan3z) [inline HappyTravel.FastpayhotelsConnector.Common/Models/Hotel/Description.cs:12]: Same for language here

File: HappyTravel.FastpayhotelsConnector.Common/Models/Hotel/HotelDetails.cs [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    /// </summary>
+    public string CheckoutHour { get; set; }
+    /// <summary>
+    /// Total number of hotel rooms
+    /// </summary>
+    public int TotalRooms { get; set; }
+    /// <summary>
+    /// Hotel time zone
+    /// </summary>
+    public string TimeZone { get; set; }
```
Comment (suhan3z) [inline HappyTravel.FastpayhotelsConnector.Common/Models/Hotel/HotelDetails.cs:56]: Time zone string?

File: HappyTravel.FastpayhotelsConnector.Common/Models/Hotel/HotelDetails.cs [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    /// </summary>
+    public DataObject HotelCategory { get; set; }
+    /// <summary>
+    /// Zone
+    /// </summary>
+    public TuristicArea Zone { get; set; }
+    /// <summary>
+    /// Hotel tags
+    /// </summary>
+    public List<object> Tags { get; set; }
```
Comment (suhan3z) [inline HappyTravel.FastpayhotelsConnector.Common/Models/Hotel/HotelDetails.cs:28]: Is it really object? No type?

File: HappyTravel.FastpayhotelsConnector.Common/Models/Hotel/Image.cs [hunk 8]
```diff
@@ -0,0 +1,21 @@
+﻿namespace HappyTravel.FastpayhotelsConnector.Common.Models.Hotel;
+
+public class Image
+{
+    /// <summary>
+    /// Type
+    /// </summary>
+    public DataObject Type { get; set; }
+    /// <summary>
+    /// Default image name in Fastpayhotels' system.
```
Comment (suhan3z) [inline HappyTravel.FastpayhotelsConnector.Common/Models/Hotel/Image.cs:10]: image name? not a url or something?

File: HappyTravel.FastpayhotelsConnector.Common/Models/Hotel/Image.cs [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+    /// </summary>
+    public string Name { get; set; }
+    /// <summary>
+    /// Define if the pic is the main one
+    /// </summary>
+    public bool Featured { get; set; }
+    /// <summary>
+    /// Url to get the image
+    /// </summary>
+    public string Url { get; set; }
```
Comment (suhan3z) [inline HappyTravel.FastpayhotelsConnector.Common/Models/Hotel/Image.cs:20]: So if we have url than we do we need name of the image?

Comment (suhan3z) [reply HappyTravel.FastpayhotelsConnector.Common/Models/Hotel/Image.cs:20]: and its better to use Uri for Urls

File: HappyTravel.FastpayhotelsConnector.Updater/Models/HotelListRequest.cs [hunk 10]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Globalization;
+
+namespace HappyTravel.FastpayhotelsConnector.Updater.Models;
+
+public class HotelListRequest
+{
+    public HotelListRequest(DateTimeOffset fromLastUpdateDate)
+    {
+        FromLastUpdateDate = fromLastUpdateDate.ToString("yyyy-MM-ddTHH\\:mm\\:ss.fffffffzzz", CultureInfo.InvariantCulture);
```
Comment (suhan3z) [inline HappyTravel.FastpayhotelsConnector.Updater/Models/HotelListRequest.cs:9]: TO constants

**Review by suhan3z — CHANGES_REQUESTED** [review]
According to comments

File: HappyTravel.FastpayhotelsConnector.Common/Models/Hotel/HotelDetails.cs (hunk 6 above)
Comment (PKorovinPromo-Z) [reply HappyTravel.FastpayhotelsConnector.Common/Models/Hotel/HotelDetails.cs:56]: Yes, the supplier returns the time zone as a string. Example: Romance Standard Time.

**Review by suhan3z — APPROVED** [review]

**Pull request merged**